### DISCUSSION TITLE
Cloudflare Zaraz Integration

### DISCUSSION THREAD

**Discussion by vittor1o** [plausible/analytics#1562] [Anything Else]
Would be great to have official guidance on how to integrate Plausible with Cloudflare Zaraz.
- https://developers.cloudflare.com/zaraz/
- https://blog.cloudflare.com/why-cloudflare-bought-zaraz/

**Comment by bjesus** [comment]
Hi Plausible team! Yo'av from the Cloudflare Zaraz team. Always really loved what you're doing - let me know if you want to work together on creating this integration.

**Reply by metmarkosaric** [reply]
thanks @vittor1o and @bjesus! we have limited development resources so cannot put effort into creating integrations for all the individual platforms as there are so many. we focus our development resources on improving our product and infrastructure instead. and then we rely on our community and people working for the different services to help us build integrations and/or write instructions for them. how's Cloudflare Zaraz different from the instructions we have on [how to proxy Plausible through Cloudflare workers](https://plausible.io/docs/proxy/guides/cloudflare)?

**Reply by chris-schra** [reply]
@metmarkosaric CF customer here - Zaraz is a plug‘n‘play service from CF so we as customers can just tick a checkbox to send data to Plausible (as well as other services)

**Comment by rmens** [comment]
Would love this as CF customer. I think integration with Cloudflare Zaraz is a good opportunity for Plausible to get free marketing in a mayor CDN platform too!

**Comment by matt08** [comment]
Any news abut that? @bjesus

**Reply by bjesus** [reply]
We are not working on this. However, we created the Managed Components standard so that vendors and community members can create their own tool implementations. See managedcomponents.dev for more information.

**Reply by sptv2reg** [reply]
Is it possible to use Zaraz with something like Plausible without getting into significant coding (as it seems from the github repos on managedcomponents)?

**Reply by bjesus** [reply]
I have zero knowledge of how Plausible works. Zaraz let's you configure the payload for an HTTP request through the UI - possibly that's enough for integrating with Plausible. Otherwise, one would have to code a Managed Component. I wouldn't say it's significant coding task, but it's definitely still coding.

**Reply by sptv2reg** [reply]
Thanks. It does have an extremely simple API so it might be possible:

```
curl -i -X POST https://plausible.io/api/event \
  -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.121 Safari/537.36 OPR/71.0.3770.284' \
  -H 'X-Forwarded-For: 127.0.0.1' \
  -H 'Content-Type: application/json' \
  --data '{"name":"pageview","url":"http://dummy.site","domain":"dummy.site","screen_width":1666}'
```

**Reply by bjesus** [reply]
We do not currently support setting headers, so unless you can pass the IP and the User Agent in the payload, you might run into troubles. It's easy to solve with a custom Cloudflare Worker, but it would still require a bit of programing.

**Reply by metmarkosaric** [reply]
we've already created a Cloudflare Worker and published the exact code needed. that works all fine for our proxy feature. see https://plausible.io/docs/proxy/guides/cloudflare. not sure if it's possible to reuse that for the Zaraz integration too?

**Reply by drush** [reply]
@metmarkosaric 
Marko, please contact your CF contacts and @bjesus about this integration.  Plausible is missing out on a huge channel opportunity here to be a one-click install on EVERY cloudflare site.  They reached out to you, so you have access here.  Please take advantage of it.  This is the screen you should be on:

![screenshot-Zaraz  Tools Configuration  Third-party tools  elevate-global co  Elevate Global  Cloudflare-Safari-2024-10-23](https://github.com/user-attachments/assets/abc229d8-1f9d-4cfe-9892-a1244af25aaa)

As you can see, there are few Analytics options currently, which will expose Plausible to a huge audience.  Moreover Zaraz has significant security, privacy and implementation advantages over GTM, which I predict means it will gain significant traction this year and next.
